migration/src/bin/migrate_to.rs

```diff
@@ -0,0 +1,393 @@
+                    Ok(())
+                }
+                async fn down(&self, manager: &SchemaManager) -> Result<(), DbErr> {
+                    manager
+                        .drop_table(Table::drop().table($table_name::Table).to_owned())
+                        .await
+                }
+            }
+
+            #[derive(Iden)]
+            enum $table_name {
+                Table,
+                Id,
+            }
+        };
+    }
+    test_migration!(m20230101_000000_test_migration_1, TestTable1);
+    test_migration!(m20230201_000000_test_migration_2, TestTable2);
+    test_migration!(m20230301_000000_test_migration_3, TestTable3);
+    test_migration!(m20230401_000000_test_migration_4, TestTable4);
+    test_migration!(m20230501_000000_test_migration_5, TestTable5);
+
+    struct TestMigrator;
+
+    #[async_trait::async_trait]
+    impl MigratorTrait for TestMigrator {
+        fn migrations() -> Vec<Box<dyn MigrationTrait>> {
+            vec![
+                Box::new(m20230101_000000_test_migration_1),
+                Box::new(m20230201_000000_test_migration_2),
+                Box::new(m20230301_000000_test_migration_3),
+                Box::new(m20230401_000000_test_migration_4),
+                Box::new(m20230501_000000_test_migration_5),
+            ]
+        }
+    }
+
+    fn all_migrations() -> Vec<&'static str> {
+        vec![
+            "m20230101_000000_test_migration_1",
+            "m20230201_000000_test_migration_2",
+            "m20230301_000000_test_migration_3",
+            "m20230401_000000_test_migration_4",
+            "m20230501_000000_test_migration_5",
+        ]
+    }
+
+    async fn test_database() -> DatabaseConnection {
+        Database::connect(ConnectOptions::new("sqlite::memory:".to_string()))
+            .await
+            .unwrap()
+    }
+
+    async fn applied_migrations(db: &DatabaseConnection) -> Vec<String> {
+        seaql_migrations::Entity::find()
+            .order_by_asc(seaql_migrations::Column::Version)
+            .all(db)
+            .await
+            .unwrap()
+            .into_iter()
+            .map(|m| m.version)
+            .collect()
+    }
+
+    #[async_std::test]
+    async fn migrate_to_up() {
+        {
+            let db = test_database().await;
+
+            // To latest
+            migrate_to::<TestMigrator>(
+                &db,
+                false,
+                Direction::Up,
+                "m20230501_000000_test_migration_5",
+            )
+            .await
+            .unwrap();
+            assert_eq!(applied_migrations(&db).await, all_migrations());
+
+            // Ensure no-op
+            migrate_to::<TestMigrator>(
+                &db,
+                false,
+                Direction::Up,
+                "m20230501_000000_test_migration_5",
+            )
+            .await
+            .unwrap();
+            assert_eq!(applied_migrations(&db).await, all_migrations());
+        }
+        {
```

Comment:
are these distinct tests?

Reply:
I wasn't originally thinking of them as distinct, but at another glance they should be.